docs/sphinx/developers/cpp/tutorial.txt

```diff
@@ -0,0 +1,235 @@
+  OME-XML data model schema, and a tree of the model objects acts as a
+  representation of the OME data model which may be modified and
+  manipulated.  The model objects may be created from an OME-XML text
+  document, and vice versa.
+
+For the simplest cases of reading and writing image data, the core
+metadata interface will likely be sufficient.  If specific individual
+parameters from the original file format are needed, then original
+metadata may also be useful.  For more advanced processing and
+rendering, the metadata store should be the next source of
+information, for example to get information about the image scale,
+stage position, instrument setup including light sources, light paths,
+detectors etc., and access to plate/well information, regions of
+interest etc.  Direct access to the OME-XML data model objects is an
+alternative to the metadata store, but is more difficult to use;
+certain modifications to the data model may only be made via direct
+access to the model objects, otherwise the higher-level metadata store
+interface should be preferred.
+
+Core metadata
+^^^^^^^^^^^^^
+
+Core metadata is accessible through the getter methods in the
+:cpp:class:`FormatReader` interface.  These operate on the *current*
+series, set using the :cpp:func:`setSeries` method.  The
+:cpp:class:`CoreMetadata` objects are also accessible directly using
+the :cpp:class:`getCoreMetadataList` method.  The
+:cpp:class:`FormatReader` interface should be preferred; the objects
+themselves are more of an implementation detail at present.
+
+.. seealso::
+
+  - :doxygen:`CoreMetadata <classome_1_1bioformats_1_1CoreMetadata.html>`
+  - :doxygen:`FormatReader <classome_1_1bioformats_1_1FormatReader.html>`
+
+
+Original metadata
+^^^^^^^^^^^^^^^^^
+
+Original metadata is stored in two forms: in a
+:cpp:class:`MetadataMap` which is accessible through the
+:cpp:class:`FormatReader` interface, which offers access to individual
+keys and the whole map for both global and series metadata.  It is
+also accessible using the metadata store; original metadata is stored
+as an :cpp:class:`XMLAnnotation`.
+
+.. seealso::
+
+  - :doxygen:`MetadataMap <classome_1_1bioformats_1_1MetadataMap.html>`
+  - :doxygen:`FormatReader <classome_1_1bioformats_1_1FormatReader.html>`
+  - :doxygen:`OriginalMetadataAnnotation <classome_1_1xml_1_1model_1_1OriginalMetadataAnnotation.html>`
+
+Metadata store
+^^^^^^^^^^^^^^
+
+Access to metadata is provided via the :cpp:class:`MetadataStore` and
+:cpp:class:`MetadataRetrieve` interfaces.  These provide setters and
+getters, respectively, to store and retrieve metadata from an
+underlying abstract metadata store.  The primary store is the
+:cpp:class:`OMEXMLMetadata` which stores the metadata in OME-XML data
+model objects (see below), and implements both interfaces.  However,
+other storage classes are available, and may be used to filter the
+stored metadata, combine different stores, or do nothing at all.
```

Comment:
Yes, it's `DummyMetadata`.  I didn't want to go into detail in this part of the docs though.